src/services/fhirService.ts

```diff
@@ -381,6 +383,16 @@ export const prepareSearchParameters = (params: IParams) => {
     delete parameters.annotations;
   }
 
+  if (params.exclude_tags) {
+    const excludeTags = taggingUtils.generateCustomTags(params.exclude_tags);
+    parameters.exclude_tags = excludeTags;
+  }
+
+  if (params.exclude_flags) {
+    parameters.exclude_tags = [...new Set([...(params.exclude_tags || []), ...params.exclude_flags])];
+    delete parameters.exclude_flags
+  }
```

Comment:
The `flag=appdata` that was passed along for count and fetchResources was being treated as a custom tag, therefore making it `custom=flag%3Dappdata`, which then obviously didn't work for filtering.

My solution to this is adding additional `exclude_flags`, which just get appended after the custom tags are handled.